ISISNeutronMuon/MDANSE
# Add logging to MDANSE and general refactoring

**Description of work**

- Replaced all print and icecream calls with calls to logging - icecream dependency has now been removed.
- Added a new option in the outputfiles configurations so that they can now output log files.
- Remove unused imports.
- Updated some remaining file headers with the current version.
- Added logs textbox in the running jobs. This outputs INFO level or higher. Warning and errors are coloured.
- Generalised the TriplePanel to MultiPanel
- Updated checkablecombobox in the file selection widget. This is now the same as the checkablecombobox in the atom selection widget.

![image](https://github.com/ISISNeutronMuon/MDANSE/assets/14276033/77c5c90f-22dd-4a21-be38-297171e5c72e)

![image](https://github.com/ISISNeutronMuon/MDANSE/assets/14276033/ab8c1ec5-41a8-46ab-9d68-403c0ac81ce1)

![image](https://github.com/ISISNeutronMuon/MDANSE/assets/14276033/61a09bcc-e895-4293-ba68-b599655d533a)


**To test**
Run a converter job with the log level selected. A log file should be outputted in the same directory as the MDT file but with a .log extension. Do the same with an analysis job. Some jobs might not output any logs. ASE and DOS produce some logs. Check that the logs are updated in the running tab. 

Try rerunning the jobs with "no logs" selected, no logs should be saved.

Also, check with the multicore setting.
